builder/builder-next/adapters/containerimage/pull.go

```diff
@@ -229,14 +229,13 @@ func (is *Source) ResolveImageConfig(ctx context.Context, ref string, opt source
 	case resolver.ResolveModePreferLocal:
 		img, err := is.resolveLocal(ref)
 		if err == nil {
-			if opt.Platform != nil && !platformMatches(img, opt.Platform) {
-				log.G(ctx).WithField("ref", ref).Debugf("Requested build platform %s does not match local image platform %s, checking remote",
-					path.Join(opt.Platform.OS, opt.Platform.Architecture, opt.Platform.Variant),
-					path.Join(img.OS, img.Architecture, img.Variant),
-				)
-			} else {
+			if opt.Platform == nil || platformMatches(img, opt.Platform) {
 				return "", img.RawJSON(), err
 			}
+			log.G(ctx).WithField("ref", ref).Debugf("Requested build platform %s does not match local image platform %s, checking remote",
+				path.Join(opt.Platform.OS, opt.Platform.Architecture, opt.Platform.Variant),
+				path.Join(img.OS, img.Architecture, img.Variant),
+			)
 		}
 		// fallback to remote
 		return is.resolveRemote(ctx, ref, opt.Platform, sm, g)
```

Comment:
I'm slightly tempted to flip the whole condition above instead of using `err == nil`;

```go
img, err := is.resolveLocal(ref)
if err != nil {
	// fallback to remote
	return is.resolveRemote(ctx, ref, opt.Platform, sm, g)
}
```

I'm curious though if such a check should check what kind of error we received, i.e., should it check if it was a "not found" error? Because there's also an error return if the given platform was invalid / could not be normalised;
https://github.com/moby/moby/blob/4bcb850902cd5d94b1c54bf6a01e12f1b73c70a5/builder/builder-next/adapters/containerimage/pull.go#L156-L160